src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1114,6 +1114,34 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		const font = this.xterm.getFont();
 		const lineHeight = font?.charHeight ? font.charHeight * font.lineHeight + 'px' : '';
 		this._accessibilityBuffer.style.lineHeight = lineHeight;
+		const commands = this.capabilities.get(TerminalCapability.CommandDetection)?.commands;
+		if (!commands?.length) {
+			const noContent = document.createElement('div');
+			const noContentLabel = nls.localize('terminal.integrated.noContent', "No terminal content available for this session.");
+			noContent.ariaLabel = noContentLabel;
+			noContent.textContent = noContentLabel;
+			this.xterm.raw.setAccessibilityBufferElements([noContent]);
```

Comment:
Is it possible for this API to be the other way around? Like `registerBufferElementProvider()` such that xterm.js requests the elements only when needed?